skbio/io/format/fastq.py

```diff
@@ -313,52 +314,38 @@ def _fastq_sniffer(fh):
     #   file. Read up to 10 records. If at least one record is read (i.e. the
     #   file isn't empty) and the quality scores are in printable ASCII range,
     #   assume the file is FASTQ.
-    if _too_many_blanks(fh, 5):
-        return False, {}
-
+    blanks, consumed = _too_many_blanks(fh, 5)
+    if blanks:
+        return False, {}, consumed
+    if not fh.seekable():
+        fh = itertools.chain(consumed, fh)
+
+    consumed = []
+    num_records = 10
+    empty = True
     try:
-        not_empty = False
-        for _, seq in zip(range(10), _fastq_to_generator(fh, phred_offset=33)):
-            split_length = len(
-                (seq.metadata["id"] + seq.metadata["description"]).split(":")
-            )
-            description = seq.metadata["description"].split(":")
+        # Scan 10 lines of the file and make sure they're valid
+        fastqgen = _parse_fastq_raw(fh, FASTQFormatError, phred_offset=33)
+        for _, seq in zip(range(num_records), fastqgen):
+            consumed += seq[4]
```

Comment:
same here with the separation of concerns -- the sniffer ideally does not need to be responsible for what it's consumed as that can be managed higher in the stack